lib/aggregate.js

```diff
@@ -87,6 +87,24 @@ function Aggregate(pipeline, model) {
 
 Aggregate.prototype.options;
 
+/**
+ * Returns default options for this aggregate.
+ *
+ * @param {Model} model
+ * @api private
+ */
+
+Aggregate.prototype._optionsForExec = function() {
+  const options = clone(this.options || {});
```

Comment:
`clone()` shouldn't be necessary here, would be ideal to avoid cloning for performance purposes

Reply:
The previous implementation also used clone(), so I retained it to ensure no existing functionality was unintentionally affected. However, I agree that removing clone() could improve performance. If it’s deemed unnecessary, I suggest calling the function directly in the constructor to initialize the options instead.